modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/EPDFrameBuffer.java

```diff
@@ -571,20 +586,57 @@ ByteBuffer getOffscreenBuffer() {
          * "QuantumRenderer modifies buffer in use by JavaFX Application Thread"
          * <https://bugs.openjdk.java.net/browse/JDK-8201567>.
          */
-        int size = xresVirtual * yresVirtual * Integer.SIZE;
+        int size = xresVirtual * yres * Integer.BYTES;
```

Comment:
This is the only problem fixed by this commit that also exists in JavaFX 13. It results in an off-screen buffer that is eight times larger than necessary when using an 8-bit or 16-bit frame buffer. The fix is to use the number of bytes per pixel (`Integer.BYTES`), not the number of bits per pixel (`Integer.SIZE`), when calculating the capacity of the 32-bit off-screen byte buffer.